client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -178,6 +178,11 @@ export class DomainDetailsForm extends PureComponent {
 		this.setState( { currentStep: newStep } );
 	}
 
+	getDomainNames = () =>
+		map( cartItems.getDomainRegistrations( this.props.cart ), 'meta' ).concat(
+			map( cartItems.getDomainTransfers( this.props.cart ), 'meta' )
+		);
```

Comment:
You could tighten this up a little with array spread if you wanted to (`map( [ ...registrations, ...transfers], 'meta' )`).